core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -466,19 +479,14 @@ public static void udpateUserRoleMappingWithExactParams(Connection dbConnection,
             }
             throw new UserStoreException(errorMessage, e);
         } finally {
-            if (localConnection) {
-                DatabaseUtil.closeAllConnections(dbConnection);
-            }
-            DatabaseUtil.closeAllConnections(null, ps);
+            close(dbConnection);
```

Comment:
Since the connection getting passed as a parameter to this method, it shouldn't close the connection inside this method.

Need to close the connection in the original place where it originally (create and )close the dbConnection.